### PR TITLE
Libvirt: add libvirt support and documentation

### PR DESCRIPTION
**Pull request by matthewoliver**
Still need to fix ip address assigning.

### CODE REVIEW

**Comment by matthewoliver** [comment]
Still working on getting the expected IP address

File: README.libvirt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+running for your user (unless you want to sudo everything).
+
+Install libvirtd etc. On fedora this is easy:
+	sudo dnf install @virtualization
+
+Let's make sure its running:
+	sudo systemctl start libvirtd
+	sudo systemctl enable libvirtd
+
+Make sure you're user is apart of the libvirt group:
```
Comment (tipabu) [inline README.libvirt:15]: Better as

> Make sure your user is a member of the libvirt group:

File: README.libvirt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+We need to enable read write to domain socket for the libvirt group, uncomment
+and make sure these are set in /etc/libvirt/libvirtd.conf:
+	unix_sock_group = "libvirt"
+	auth_unix_rw = "none"
+	auth_unix_ro = "none"
+
+Then restart libvirtd
+	sudo systemctl restart libvirtd
+
+# Now we need to make sure we're looking at the default URI. There are many
```
Comment (tipabu) [inline README.libvirt:27]: Two things:

- Why the leading `#`s?
- What does this *mean*? Should it ever *not* be `qemu:///system`? If so, how does a user find out the right value?

File: Vagrantfile [hunk 3]
```diff
@@ -112,6 +113,15 @@ Vagrant.configure("2") do |global_config|
         end
       end
 
+      config.vm.provider "libvirt" do |lv, override|
```
Comment (tipabu) [inline Vagrantfile:116]: Does `lv` have an equivalent of `vb.gui`? Eh, I suppose we could add it later if/when needed...

**Pull request merged**